sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -213,6 +213,20 @@ case class DataSourceResolution(
           }
           ShowTablesCommand(Some(namespace.quoted), pattern)
       }
+
+    case SetCatalogAndNamespaceStatement(catalogName, namespace) =>
+      catalogName match {
+        case Some(c) =>
+          validateCatalog(c)
+          SetCatalogAndNamespace(catalogManager, Some(c), namespace)
```

Comment:
when we set catalog and namespace together, shall we also check if the namespace exists?

Reply:
Done